### ISSUE TITLE
bug(datepicker):  It is showing truncated placeholder text on date range edit boxes in high contrast mode

### ISSUE THREAD

**Issue by zarend** [angular/components#24081]
#### Reproduction

Steps to reproduce:
Prerequisite: Enable High Contrast Black Mode (Settings > Personalization > Colors > High Contrast Settings)

1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Observe the date range edit boxes in high contrast mode
 
#### Expected Behavior
It should show complete placeholder text on date range edit boxes in high contrast mode

#### Actual Behavior
It is showing "te range" instead of "Enter date range" as placeholder text in the date range edit boxes in high contrast mode

#### Environment
OS| Browser |
Windows 10 | Firefox 90.0.2 (64-bit) |

##### Additional Notes
This is copied from an [internal bug report](http://b/195152476) found during a11y testing.

**Comment by crisbeto** [comment]
This is a known issue due to the following workaround not working in Firefox: https://github.com/angular/components/blob/master/src/material/form-field/form-field.scss#L61. Ideally we would solve this by not using a `border` for spacing in the form field, but the last time we tried removing it, it messed up the baseline.

**Comment by zarend** [comment]
I cannot reproduce, and I believe this was fixed by #11736
